### ISSUE TITLE
jsonschema `$ref` parsed as yaml?

### ISSUE THREAD

**Issue by freol35241** [koxudaxi/datamodel-code-generator#564]
**Describe the bug**
`$ref`s in jsonschemas seems to be parsed as yaml? I get the following error when the parser tries to follow the `$ref`:

```
Traceback (most recent call last):
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/__main__.py", line 495, in main
    generate(
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/__init__.py", line 353, in generate
    results = parser.parse()
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/base.py", line 426, in parse
    self.parse_raw()
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1177, in parse_raw
    self._parse_file(self.raw_obj, obj_name, path_parts)
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1250, in _parse_file
    self.parse_obj(obj_name, root_obj, path_parts or ['#'])
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1142, in parse_obj
    self.parse_ref(obj, path)
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1084, in parse_ref
    self.parse_ref(value, path)
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1064, in parse_ref
    self.resolve_ref(obj.ref)
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1054, in resolve_ref
    self._get_ref_body(relative_path),
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1005, in _get_ref_body
    return self._get_ref_body_from_url(resolved_ref)
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1010, in _get_ref_body_from_url
    return self.remote_object_cache.get_or_put(
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/__init__.py", line 26, in get_or_put
    value = self[key] = default_factory(key)
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1011, in <lambda>
    ref, default_factory=lambda key: load_yaml(self._get_text_from_url(key))
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/datamodel_code_generator/__init__.py", line 85, in load_yaml
    return yaml.load(stream, Loader=SafeLoader)
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/home/freol/dev/brefv/venv/lib/python3.8/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "yaml/_yaml.pyx", line 707, in yaml._yaml.CParser.get_single_node
  File "yaml/_yaml.pyx", line 726, in yaml._yaml.CParser._compose_document
  File "yaml/_yaml.pyx", line 905, in yaml._yaml.CParser._parse_next_event
yaml.scanner.ScannerError: mapping values are not allowed in this context
  in "<unicode string>", line 9, column 25
```

**To Reproduce**
When trying to generate code for the following example jsonschema (including a reference):

Example schema:
```json
{
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "type": "object",
    "title": "Position",
    "description": "A reported position as a GeoJSON Point",
    "required": [
      "position"
    ],
    "properties": {
      "position": {
        "$ref": "http://geojson.org/schema/Point.json"
      }
    },
    "additionalProperties": false
}
```

Used commandline:
```
$ datamodel-codegen --input position.json --input-file-type jsonschema --output model.py
```

**Expected behavior**
Successful generation of a pydantic model.

**Version:**
 - OS: Ubuntu 20.04 (WSL2 under win10)
 - Python version: 3.8
 - datamodel-code-generator version: 0.11.14

**Additional context**

**Comment by freol35241** [comment]
Thanks for the answer @koxudaxi . It made me dig a bit deeper and this is actually quite interesting, it seems that the yaml parser is thrown off by different indentation in the JSON-files attached to this post. The first one works for me whereas the second one does not. I was under the belief that json is not sensitive to indentation?

[first_and_second.zip](https://github.com/koxudaxi/datamodel-code-generator/files/7299559/first_and_second.zip)

**Comment by kylebarron** [comment]
I also hit this today. A simple example is a list of GeoJSON Features.

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "https://example.com/list-of-features.schema.json",
  "title": "ListOfFeatures",
  "type": "object",
  "required": ["features"],
  "additionalProperties": false,
  "properties": {
    "features": {
      "type": "array",
      "items": {
        "$ref": "https://geojson.org/schema/Feature.json"
      }
    }
  }
}
```

If I run `datamodel-codegen` on this, it fails:

```
> datamodel-codegen \
  --input list-of-features.schema.json \
  --input-file-type jsonschema \
  --output test.py
Traceback (most recent call last):
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/__main__.py", line 495, in main
    generate(
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/__init__.py", line 353, in generate
    results = parser.parse()
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/base.py", line 426, in parse
    self.parse_raw()
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1177, in parse_raw
    self._parse_file(self.raw_obj, obj_name, path_parts)
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1250, in _parse_file
    self.parse_obj(obj_name, root_obj, path_parts or ['#'])
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1142, in parse_obj
    self.parse_ref(obj, path)
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1084, in parse_ref
    self.parse_ref(value, path)
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1067, in parse_ref
    self.parse_ref(obj.items, path)
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1064, in parse_ref
    self.resolve_ref(obj.ref)
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1054, in resolve_ref
    self._get_ref_body(relative_path),
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1005, in _get_ref_body
    return self._get_ref_body_from_url(resolved_ref)
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1010, in _get_ref_body_from_url
    return self.remote_object_cache.get_or_put(
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/__init__.py", line 26, in get_or_put
    value = self[key] = default_factory(key)
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1011, in <lambda>
    ref, default_factory=lambda key: load_yaml(self._get_text_from_url(key))
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/datamodel_code_generator/__init__.py", line 85, in load_yaml
    return yaml.load(stream, Loader=SafeLoader)
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/Users/kbarron/.pyenv/versions/3.8.10/lib/python3.8/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "yaml/_yaml.pyx", line 707, in yaml._yaml.CParser.get_single_node
  File "yaml/_yaml.pyx", line 726, in yaml._yaml.CParser._compose_document
  File "yaml/_yaml.pyx", line 905, in yaml._yaml.CParser._parse_next_event
yaml.scanner.ScannerError: mapping values are not allowed in this context
  in "<unicode string>", line 11, column 25
```

If I apply the diff:
```diff
+   "$id": "blank",
-   "$id": "https://example.com/list-of-features.schema.json",
```

Then `datamodel-codegen` works as expected. My read of the JSON Schema docs only says that an `$id` must be unique, not that it actually exists. It's hard to find explicit, official documentation on this, but for example [this SO answer](https://stackoverflow.com/a/67674405) says

> Note, these are NOT URLs, and are not nessecerily relating to a location of any physical files, nor are they nessecerily network-addressable. `$id` is an identifier only.

**Comment by koxudaxi** [comment]
@kylebarron
I have released a new version 0.11.15

**Comment by freol35241** [comment]
@koxudaxi @kylebarron  thanks, just tested 0.11.15 and can confirm this solves my issue 👍
